pkg/common/log/logger.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// GetServiceLogger returns service logger, it will using the service as the logger name, and
+// append FieldNameServiceUUID field to the logger
+func GetServiceLogger(logger *zap.Logger, service ServiceType, uuid string) *zap.Logger {
+	return logger.Named(string(service)).With(zap.String(FieldNameServiceUUID, uuid))
+}
+
+// GetModuleLogger returns the module logger, it will add ".module" to logger name.
+// e.g. if the logger's name is cn-service, module is txn, the new logger's name is
+// "cn-service.txn".
+func GetModuleLogger(logger *zap.Logger, module Module) *zap.Logger {
+	return logger.Named(string(module))
+}
+
+// Log is the entry point for mo log printing. Return true to indicate that the log
+// is being recorded by the current LogContext.
+func (c LogContext) Log(msg string, fields ...zap.Field) bool {
+	if c.logger == nil {
+		panic("missing logger")
+	}
+
+	for _, fiter := range filters {
+		if !fiter(c) {
+			return false
+		}
+	}
+
+	if ce := c.logger.Check(c.level, msg); ce != nil {
+		if len(c.fields) == 0 {
+			c.fields = fields
+		} else {
+			c.fields = append(c.fields, fields...)
+		}
+
+		ce.Write(c.fields...)
```

Comment:
we should add field like `trace.ContextField(ctx)` by default.
If ctx contains span, log_info can save it in `log_info.trace_id`, `log_info.span_id` column.